MSAL/src/MSALResult.m

```diff
@@ -94,12 +101,26 @@ + (MSALResult *)resultWithTokenResult:(MSIDTokenResult *)tokenResult
     
     NSString *tenantId = claims.realm;
 
-    MSALAccount *account = [[MSALAccount alloc] initWithUsername:resultAccount.username
-                                                            name:resultAccount.name
-                                                   homeAccountId:resultAccount.accountIdentifier.homeAccountId
-                                                  localAccountId:resultAccount.localAccountId
-                                                     environment:tokenResult.authority.environment
-                                                        tenantId:tenantId];
+    MSALAccount *account;
+    if (cache)
+    {
+        MSALAccountsProvider *accountProvider = [[MSALAccountsProvider alloc] initWithTokenCache:cache clientId:tokenResult.accessToken.clientId];
+        
+        if (accountProvider)
+        {
+            NSError *localError;
+            account = [accountProvider accountForHomeAccountId:resultAccount.accountIdentifier.homeAccountId error:&localError];
```

Comment:
I'm not sure why do you need to do an additional cache lookup in here?

Reply:
discussed offline, we are going to return nil tenantProfiles in this case. I am going to change it :)